dialogue-core/src/main/java/com/palantir/dialogue/core/LimitedChannelToChannelAdapter.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.dialogue.core;
+
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.dialogue.Channel;
+import com.palantir.dialogue.Endpoint;
+import com.palantir.dialogue.Request;
+import com.palantir.dialogue.Response;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.exceptions.SafeRuntimeException;
+import java.util.function.Supplier;
+
+/** Adapter from {@link LimitedChannel} to {@link Channel} which produces a failed future when results are limited. */
+final class LimitedChannelToChannelAdapter implements Channel {
+
+    // Avoid method reference allocations
+    @SuppressWarnings("UnnecessaryLambda")
+    private static final Supplier<ListenableFuture<Response>> limitedResultSupplier =
+            () -> Futures.immediateFailedFuture(new SafeRuntimeException("Failed to make a request"));
```

Comment:
I think we're going to need to give people more information in this exception message (or possibly a comment here) otherwise we're going to be fielding questions in `#dev-foundry-infra` from people asking "why dialogue is breaking them" and telling them they can't make requests ;)

When people hit this, we probably don't want them to do their own retrying right?  Perhaps we should even put some instrumentation on this, so we know exactly how many times our client refused to send a request out the door?

Reply:
Agreed, I like the idea of replacing the limited-channel result of `Optional<future<response>>` with a union of `limited-reason` or `future<response>`. Also like the idea of forcefully attempting a request through the limiter if all nodes have limited our result to avoid purely client-side badness.